chapter08/ard/ex1.pl

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/perl -w
+use strict;
+use warnings;
+use 5.010;
+
+# use all of the IO modules!
+use Time::Piece;
+use IO::Handle;
+use IO::Scalar;
+use IO::Tee;
+
+# nasty globals, but will do for now
+my $output_handle; # this will ultimately point to our destination(s)
+my $scalar;
+
+my $t = localtime;
+my $output_string = sprintf( "Today is %s %u %s %u", $t->fullday, $t->day_of_month, $t->fullmonth, $t->year );
+
+
+print "Output to file, scalar or both? [fsb] ";
+chomp( my $choice = lc( <STDIN> ) );
+#$choice = lc( $choice );
+
+my %dispatch = (
+  f => \&do_file,
+  s => \&do_scalar,
+  b => \&do_both,
+);
+
+( $dispatch{$choice} || \&do_invalid )->(); 
```

Comment:
Why not return the handle from these functions rather than assigning to the global?